examples/first/main.go

```diff
@@ -16,30 +16,26 @@ import (
 	"github.com/beatlabs/patron/log"
 )
 
+const (
+	name    = "first"
+	version = "1.0.0"
+)
+
 func init() {
 	err := os.Setenv("PATRON_LOG_LEVEL", "debug")
 	if err != nil {
-		fmt.Printf("failed to set log level env var: %v", err)
-		os.Exit(1)
+		log.Fatalf("failed to set log level env var: %v", err)
 	}
+
 	err = os.Setenv("PATRON_JAEGER_SAMPLER_PARAM", "1.0")
 	if err != nil {
-		fmt.Printf("failed to set sampler env vars: %v", err)
-		os.Exit(1)
+		log.Fatalf("failed to set sampler env vars: %v", err)
 	}
 }
 
 func main() {
-	name := "first"
-	version := "1.0.0"
-
-	err := patron.SetupLogging(name, version)
```

Comment:
The only thing I would argue that it's being set up in `Run` and for me, it looks like a bit inconsistent API.

It would make more sense if `nilLogger` ( a default implementation ) was panicking without set logger and the only way to set it up would be calling `patron.SetupLogging` before `Run`. That looks consistent.

The current API inconsistent a bit and allows skipping  `patron.SetupLogging` without any "penalties"

WDYT?